References are formatted properly #2837

Inny05 opened the pull request:
Fixes #2837


Changes made in this Pull Request:
A docstring was added to line 257 so as to properly cite the reference.

Inny05 committed:
Update contacts.py

IAlibay commented on package/MDAnalysis/analysis/contacts.py:
```diff
@@ -254,7 +254,7 @@ def soft_cut_q(r, r0, beta=5.0, lambda_constant=1.8):
       fraction of native contacts
 
-    References
+    """ References
```
I'm not sure what the intent here is, but this breaks the existing docstring.

lilyminium replied:
@Inny05 this change breaks tests and documentation, as you can see from the failing test. I'd suggest removing it before altering the documnentation -- if docs can build, we can see the effect of new changes.

IAlibay reviewed (changes requested):
Thanks for looking to tackle this issue @Inny05

In this case my understanding of the issue is that the docstring is no broken but that the formatting for the reference is broken.

@lilyminium's suggestion here was that we should be using bibtext to create a file containing all our references and then load them into our documentation as required. I would suggest having a look at the userguide, this might be of particular interest: https://github.com/MDAnalysis/UserGuide/blob/53079fa48f7d04209a3cf40633303c8d91a482f1/doc/source/scripts/clean_example_notebooks.py#L51

Inny05 commented:
Thank you for your response, I will make the necessary changes.

Regards

On Wed, 30 Mar 2022, 17:01 Irfan Alibay, ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> Thanks for looking to tackle this issue @Inny05
> <https://github.com/Inny05>
>
> In this case my understanding of the issue is that the docstring is no
> broken but that the formatting for the reference is broken.
>
> @lilyminium <https://github.com/lilyminium>'s suggestion here was that we
> should be using bibtext to create a file containing all our references and
> then load them into our documentation as required. I would suggest having a
> look at the userguide, this might be of particular interest:
> https://github.com/MDAnalysis/UserGuide/blob/53079fa48f7d04209a3cf40633303c8d91a482f1/doc/source/scripts/clean_example_notebooks.py#L51
> ------------------------------
>
> In package/MDAnalysis/analysis/contacts.py
> <https://github.com/MDAnalysis/mdanalysis/pull/3585#discussion_r838514409>
> :
>
> > @@ -254,7 +254,7 @@ def soft_cut_q(r, r0, beta=5.0, lambda_constant=1.8):
>      Q : float
>        fraction of native contacts
>
> -    References
> +    """ References
>
> I'm not sure what the intent here is, but this breaks the existing
> docstring.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MDAnalysis/mdanalysis/pull/3585#pullrequestreview-926077312>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APJTTAFST6GKU3FSKZFFUYTVCR3GJANCNFSM5SBXPZNA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

ojeda-e commented:
Thanks for working on this @Inny05. 

Would you please also add the issue number associated with this PR? You can edit [your initial comment](https://github.com/MDAnalysis/mdanalysis/pull/3585#issue-1186373539) in the line that says  Fixes #_number_. You are fixing issue #2837, I think.

By adding the number, GitHub will automatically link this PR to the respective issue and also helps reviewers to identify the issue you are addressing.

ojeda-e commented on package/MDAnalysis/analysis/contacts.py:
```diff
@@ -254,7 +254,7 @@ def soft_cut_q(r, r0, beta=5.0, lambda_constant=1.8):
+    """ References
     ----------
     .. [Best2013] RB Best, G Hummer, and WA Eaton, "Native contacts determine protein
        folding mechanisms in atomistic simulations" _PNAS_ **110** (2013),
```
The idea is to have this and all the references included in MDAnalysis in BibTeX format. References in the User Guide are in BibTex. See [this link](https://github.com/MDAnalysis/UserGuide/blob/develop/doc/source/references.bib#L2) as a hint.

Inny05 replied:
I have created a bibtex file using Texmaker. I need help getting it in the repository. I was considering citing the repository, do you think that would solve it?

lilyminium replied:
I'm not sure what you mean by citing the repository. We don't need to here,  we are citing Best and Franklin. You also seem to have added your file into the repository already?

Inny05 commented:
Thank you for clarifying this. I am currently working on resolving it.

Regards

On Wed, Mar 30, 2022 at 6:42 PM Estefania Barreto-Ojeda <
***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In package/MDAnalysis/analysis/contacts.py
> <https://github.com/MDAnalysis/mdanalysis/pull/3585#discussion_r838807958>
> :
>
> > @@ -254,7 +254,7 @@ def soft_cut_q(r, r0, beta=5.0, lambda_constant=1.8):
>      Q : float
>        fraction of native contacts
>
> -    References
> +    """ References
>      ----------
>      .. [Best2013] RB Best, G Hummer, and WA Eaton, "Native contacts determine protein
>         folding mechanisms in atomistic simulations" _PNAS_ **110** (2013),
>
> The idea is to have this and all the references included in MDAnalysis in
> BibTeX format. References in the User Guide are in BibTex. See this link
> <https://github.com/MDAnalysis/UserGuide/blob/develop/doc/source/references.bib#L2>
> as a hint.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MDAnalysis/mdanalysis/pull/3585#pullrequestreview-926497464>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APJTTADO6LBX47COLVC2E6LVCSHB5ANCNFSM5SBXPZNA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Inny05 committed:
Add files via upload

Inny05 committed:
References are formatted properly #2837

A bibliography file was created for the reference in line 257 and 315 properly formatted in this bibtex file

lilyminium commented on package/MDAnalysis/analysis/md.bib:
```diff
@@ -0,0 +1,23 @@
+@article{[Best2013],
```
I'd suggest not using unnecessary punctuation in the citation key, I feel like it could confuse the extension.

lilyminium commented on package/MDAnalysis/analysis/md.bib:
```diff
@@ -0,0 +1,23 @@
+@article{[Best2013],
+		author= {RB Best, G Hummer, and WA Eaton,},
+		title = {"Native contacts determine protein folding mechanisms in atomistic simulations"}, 				
+		 _PNAS_ **110** 
```
This is not part of the bibtex entry which needs to be formatted in a `field = {value}` style. It might be easier to click on the DOI URL and just download a bibtex entry from there.

lilyminium reviewed (changes requested):
Hi @Inny05, thanks for giving this a go. Shifting the docs onto using bibtex requires using the [sphinxcontrib-bibtex](https://sphinxcontrib-bibtex.readthedocs.io/en/latest/usage.html) extension and formatting the bib file and citation the way they require you to. Using sphinxcontrib-bibtex has the advantage that we can isolate *all* references cited throughout MDAnalysis into one file, which we will probably save in the `package/docs/sphinx/source/` directory somewhere. For example, the one in the user guide is saved in `doc/sources/references.bib`: https://github.com/MDAnalysis/UserGuide/blob/develop/doc/source/references.bib

In order to tell sphinx how to use bibtex, we need to install the bibtex extension. This has a couple steps:
* adding the `sphinxcontrib-bibtex` package to the packages we use to build docs here: https://github.com/MDAnalysis/mdanalysis/blob/bf95f9cc18cafb2f387f90e467c7944b371abeb4/.github/workflows/gh-ci.yaml#L164 (just adding the `sphinxcontrib-bibtex` after `msmb_theme==1.2.0` in the quotes will do)
* Normally we would add  the `sphinxcontrib-bibtex` package to the packages we use to build docs on ReadTheDocs, for PR previews. In this file: https://github.com/MDAnalysis/mdanalysis/blob/develop/maintainer/conda/environment.yml but it appears to already be here
* adding this line https://github.com/MDAnalysis/UserGuide/blob/2554ce0f98c59930371e7ee12475cca043d4cff2/doc/source/conf.py#L87 to `conf.py` in the `package/docs/sphinx/source/` directory in *this* repo
* adding a line `bibtex_bibfiles = ['refs.bib']` (or whatever you call the `.bib` file) to `conf.py` as well

Then to actually cite a source in the documentation, you would need to use `:cite:p:` (largely). So for example, in this line:

https://github.com/Inny05/mdanalysis/blob/eeac01f318d9b28b932689a4660c78701241d952/package/MDAnalysis/analysis/contacts.py#L230

It could get updated to:
```
The native contact function is defined as :cite:p:`Best2013`.
```

ojeda-e commented:
Hi @Inny05, are you still working on this PR? There hasn't been any activity in the last three days.

If we do not get a response over the next 4 days, we will [close this PR as stale](https://github.com/MDAnalysis/mdanalysis/wiki/GSoC-FAQ#closing-stale-prs).

Inny05 commented:
Yes, I am still working on this.

On Tue, 5 Apr 2022, 19:27 Estefania Barreto-Ojeda, ***@***.***>
wrote:

> Hi @Inny05 <https://github.com/Inny05>, are you still working on this PR?
> There hasn't been any activity in the last three days.
>
> If we do not get a response over the next 4 days, we will close this PR
> as stale
> <https://github.com/MDAnalysis/mdanalysis/wiki/GSoC-FAQ#closing-stale-prs>
> .
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MDAnalysis/mdanalysis/pull/3585#issuecomment-1089157569>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APJTTAGZZCWLBOCQYBAEEJ3VDSAYBANCNFSM5SBXPZNA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Inny05 committed:
Update md.bib

I used the DOI url to get a bibtex format from google scholar

Inny05 committed:
Update on contact.py line 257 

Reference citation on line 257

micaela-matta commented:
@Inny05 since there has not been much activity here I would ask @ojeda-e to mark this as stale.

ojeda-e commented:
Thanks @micaela-matta 
There hasn't been activity on this PR for 9 days. I am closing the PR marking it as [stale](https://github.com/MDAnalysis/mdanalysis/wiki/GSoC-FAQ#closing-stale-prs).
